src/binary-search-tree.js

```diff
@@ -12,17 +12,43 @@ class BinarySearchTree {
   // assigns it to either the left or right subtree,
   // depending on its value
   insert(value) {
-
+    const node = new BinarySearchTree(value);
+    const addtoTree = (val) => {
+      if (val.value >= value) {
```

Comment:
You don't want both if statements here to be checking if `val.value` is equal to `value`. Only one if statement should be accounting for equality, not both.